### ISSUE TITLE
impl. Object#class

### ISSUE THREAD

**Issue by yhara** [shiika-lang/shiika#224]
Until pattern match is implemented, there should be a workaround to branch with object class.

```
class A; end
class B : A; end
class C : A; end
...
if o.class.name == "B"
  ...
```

**Comment by yhara** [comment]
Well no,

```
if o.class == "B"
  o.foo  # The type of `o` is still `A`, not `B`, so you cannot call methods of B
```

**Comment by yhara** [comment]
`Object#class` is implemented in #299. 

>   o.foo  # The type of `o` is still `A`, not `B`, so you cannot call methods of B

This issue(?) is still present but I'm working on pattern-matching now.
